kpack-image-builder/controllers/buildworkload_controller.go

```diff
@@ -139,6 +139,17 @@ func (r *BuildWorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
+	if len(buildWorkload.Spec.Buildpacks) > 0 {
+		// Specifying buildpacks is not supported
+		failureStatusConditionMessage := `Only the "default" buildpack is supported`
```

Comment:
I think this error message might be confusing. There's not really a concept of a "default" buildpack, right? Like there is a default builder that has a collection of buildpacks. Maybe that could be mentioned?

That's not really a CF concept though so if this error is bubbled up and displayed through `cf push`, maybe an error message that mentions only "buildpack auto-detection" is supported?

IDK, open to thoughts on this one, but I think something that mentions auto-detection and you can't specify individual buildpacks would be good.

Reply:
I guess `'default'` and `'null'` are both valid CF App Manifest concepts to empty out the buildpacks array on the Cloud Controller side. 🤔 

https://github.com/cloudfoundry/cloud_controller_ng/commit/f402ddde20703b17e805b317fcaec6a25abcd5c7

Reply:
I missed that when I was writing this story. The `cf push` command has built-in logic to turn `default` into an empty buildpacks array, so I assumed we didn't need this. Do you think we should add this? If so, I think that the CFProcess reconciler is the right place to handle turning `default` into a no-op

Reply:
Actually, we could also do the `default` translation in the App repo, the payload or the handler. As long as we don't make every builder reimplement this logic.

Reply:
I took a second look at the CCNG link above. The logic to handle `default` is only invoked when the Manifest has a `buildpack` key (not `buildpacks`). This commit doesn't support that field at all, so the logic wouldn't fit anyway. If we're concerned about supporting both `buildpack` and `buildpacks`, then I can split out a separate story to handle the singular version with the `default` logic included.

Reply:
Re: supporting the `buildpack` field in addition to `buildpacks`.

It is probably worth making a story to capture this context in case it comes up, but I don't think it's worth supporting at this time.  If we ever do it then we can handle `default`/`null`, but those were both legacy manifest values when we implemented them in CCNG in the first-place.